x-pack/plugins/cases/public/components/severity/selector.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHealth,
+  EuiSuperSelect,
+  EuiSuperSelectOption,
+} from '@elastic/eui';
+import React from 'react';
+import { CaseSeverity } from '../../../common/api';
+import { severities } from './config';
+
+interface Props {
+  selectedSeverity: CaseSeverity;
+  onSeverityChange: (status: CaseSeverity) => void;
+  isLoading: boolean;
+  isDisabled: boolean;
+}
+
+export const SeveritySelector: React.FC<Props> = ({
+  selectedSeverity,
+  onSeverityChange,
+  isLoading,
+  isDisabled,
+}) => {
+  const caseSeverities = Object.keys(severities) as CaseSeverity[];
+  const options: Array<EuiSuperSelectOption<CaseSeverity>> = caseSeverities.map((severity) => {
+    const severityData = severities[severity];
+    return {
+      value: severity,
+      disabled: isDisabled,
```

Comment:
I think is better to disable the whole `EuiSuperSelect` component and not individually each option. You can pass the `disabled={isDisabled}` prop to `EuiSuperSelect` component.